### PR TITLE
Add EquipmentPart unit tests

### PR DESCRIPTION
**Pull request by sixlettervariables**
This PR adds EquipmentPart unit tests and fixes five bugs found along the way:
1. Variable weight sizes were not saved in some circumstances.
2. Weapon Bays weren't checked for destruction in remove. Well, they weren't before due to a bug, then removed as a "dead code path". This restores it.
3. LAM bomb bays always reported 60 minutes to fix, even when not broken.
4. MissingEquipmentPart's did not apply variable sized sticker prices correctly
5. MissingEquipmentPart did not override MissingPart::clone

TODO:
- [x] EquipmentPart::checkWeaponBay Test
- [X] MissingEquipmentPart tests

### CODE REVIEW

File: MekHQ/src/mekhq/campaign/parts/equipment/MissingEquipmentPart.java [hunk 1]
```diff
@@ -178,149 +188,118 @@ public boolean isAcceptableReplacement(Part part, boolean refit) {
...
-        newPart.setUnit(unit);
-        if(part instanceof EquipmentPart) {
-            EquipmentPart eqpart = (EquipmentPart)part;
-            EquipmentType et = eqpart.getType();
-            return (type.equals(et) && getTonnage() == part.getTonnage())
-                    && (size == eqpart.getSize())
-                    && part.getStickerPrice().equals(newPart.getStickerPrice());
+        EquipmentPart newPart = getNewPart();
+        newPart.setEquipmentNum(getEquipmentNum());
+        newPart.setUnit(unit); // CAW: find a way to do this without setting a unit
```
Comment (NickAragua) [inline MekHQ/src/mekhq/campaign/parts/equipment/MissingEquipmentPart.java:193]: Do we want to un-set the unit afterwards?

**Review by NickAragua — COMMENTED** [review]
Couple of minor comments/questions.

File: MekHQ/src/mekhq/campaign/parts/equipment/MissingEquipmentPart.java (hunk 1 above)
Comment (sixlettervariables) [reply MekHQ/src/mekhq/campaign/parts/equipment/MissingEquipmentPart.java:193]: It gets garbage collected after this call. Now that I have a good test I'll probably move the variable cost calculator to a shared class and use that here instead of this tomfoolery.

File: MekHQ/src/mekhq/campaign/parts/equipment/EquipmentPart.java [hunk 2]
```diff
@@ -633,46 +633,24 @@ public boolean isOmniPoddable() {
...
+        final Mounted mounted = getMounted();
+        if ((mounted != null) && (mounted.getLocation() != Entity.LOC_NONE)) {
+            return getUnit().getEntity().getLocationName(mounted.getLocation());
         }
+
         return null;
     }
 
     @Override
     public boolean isInLocation(String loc) {
```
[the remark this answers is not in the document]
Comment (sixlettervariables) [reply MekHQ/src/mekhq/campaign/parts/equipment/EquipmentPart.java:645]: I tried my absolute damnedest to touch the least external code possible.

My last attempts at comprehensive tests seemed to have missed corner cases here and there. This time I have a much higher confidence level because I left the smelly code smelly.

**Review by NickAragua — APPROVED** [review]

**Comment by sixlettervariables** [comment]
Okay squashed it all and fixed the merge conflict @NickAragua

**Review by NickAragua — APPROVED** [review]
